redwoodjs/redwood
# fix: Support Custom Id Field Names in when generating Cells

Fixes https://github.com/redwoodjs/redwood/issues/9776

Context:

The PR https://github.com/redwoodjs/redwood/pull/8264 added support for Custom Id Fields for Prisma models in Scaffolds, but did not address or support these in Cell generation.

This PR implements this.

To verify:

1. Add models with a uuid as the primary key
```
model CustomIdField {
  uuid String @id @default(uuid())
  name String
}

model Airplane {
  uuid String @id @default(uuid())
  name String
}
```

2. Generate both singular and list cells

`yarn rw g cell Airplane`
`yarn rw g cell Airplanes`

3. Confirm that `uuid` is used in the generated cells and mocks.

For example:

```tsx
import type {
  FindAirplaneQuery2,
  FindAirplaneQuery2Variables,
} from 'types/graphql'

import type { CellSuccessProps, CellFailureProps } from '@redwoodjs/web'

export const QUERY = gql`
  query FindAirplaneQuery2($id: String!) {
    airplane: airplane(uuid: $id) {
      uuid
    }
  }
`

export const Loading = () => <div>Loading...</div>

export const Empty = () => <div>Empty</div>

export const Failure = ({
  error,
}: CellFailureProps<FindAirplaneQuery2Variables>) => (
  <div style={{ color: 'red' }}>Error: {error?.message}</div>
)

export const Success = ({
  airplane,
}: CellSuccessProps<FindAirplaneQuery2, FindAirplaneQuery2Variables>) => {
  return <div>{JSON.stringify(airplane)}</div>
}
```

and in mock:

```ts
// Define your own mock data here:
export const standard = (/* vars, { ctx, req } */) => ({
  airplane: {
    uuid: '42',
  },
})
```
